packages/date/src/index.js

```diff
@@ -522,9 +505,12 @@ export function format( dateFormat, dateValue = new Date() ) {
  * @return {string} Formatted date in English.
  */
 export function date( dateFormat, dateValue = new Date(), timezone ) {
-	return format(
-		dateFormat,
-		utcToZonedTime( dateValue, getActualTimezone( timezone ) )
+	return formatTZ(
+		offsetDate( dateValue, timezone ),
+		translateFormat( dateFormat, dateValue ),
```

Comment:
We pass an offset date to formatTZ, should we also pass the offset date to translateFormat?